# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/quota/QuotaManager.h
# Commit: 5ae0cf36c38b
# Full Hash: 5ae0cf36c38bc5d79553382de16f41a2795de86c
# Author: Jan Varga <jvarga@mozilla.com>
# Date: 2024-10-13 21:18:59
# Regressor Bug: 1866402
# File Overlap Count: 1
# Description:
#   Bug 1866402 - Add support for initializing corresponding origin to QuotaManager::OpenClientDirectory; r=devtools-reviewers,nchevobbe
#   
#   QuotaManager::OpenClientDirectory can now get initialization status of
#   corresponding origin and initialize the origin if it's needed.
#   
# ==============================================================================

diff -r 7b3ae7465c92 -r 5ae0cf36c38b dom/quota/QuotaManager.h
--- a/dom/quota/QuotaManager.h	Sun Oct 13 10:10:23 2024 +0000
+++ b/dom/quota/QuotaManager.h	Sun Oct 13 10:10:23 2024 +0000
@@ -298,7 +298,7 @@
   // must be always done explicitly before the lock object is destroyed (when
   // the last strong reference is removed).
   RefPtr<ClientDirectoryLockPromise> OpenClientDirectory(
-      const ClientMetadata& aClientMetadata,
+      const ClientMetadata& aClientMetadata, bool aCreateIfNonExistent = true,
       Maybe<RefPtr<ClientDirectoryLock>&> aPendingDirectoryLockOut = Nothing());
 
   RefPtr<ClientDirectoryLock> CreateDirectoryLock(